wordpress-mobile/WordPress-iOS
# In-App Notifications (Notices) UI

This PR adds custom UI to our new In-App Notifications (notices) system. This replaces the placeholder `UIAlertController` UI we're currently using.

![notices-1](https://user-images.githubusercontent.com/4780/34055449-16ca29d6-e1c7-11e7-98a5-5bce3ef1b437.gif)

The new `NoticeView` is presented from the bottom of the tab bar controller, and can also contain an optional action:

![simulator screen shot - iphone 8 - 2017-12-15 at 17 29 57](https://user-images.githubusercontent.com/4780/34054646-c8bc1cac-e1c3-11e7-8b00-0403e4a8e044.png)

On iPad (regular width size classes), the notice view is limited to a maximum width of 50% of the screen:

<img width="754" alt="screen shot 2017-12-15 at 17 28 39" src="https://user-images.githubusercontent.com/4780/34054667-e0d242bc-e1c3-11e7-8e04-9d8696c69f6c.png">

**Known limitations for this first version:**

* Notices are displayed in the tab bar controller's view, so will currently be obscured by modal view controllers and alert views / action sheets.
* The failure notice for media will be added in a future PR, as will the Write Post action for successful uploads.

**To test:**

* In the media library, use the new async option to upload media.
* Ensure that the notice appears at the bottom of the screen.
* Check the appearance on a range of devices and orientations.
* The notice should hide itself after 5 seconds.
* Try tapping a notice to hide it.
